src/instructlab/config/init.py

```diff
@@ -183,6 +202,59 @@ def init(
     )
 
 
+def hw_auto_detect() -> tuple[str | None, int | None, int | None, _train, bool]:
+    # Third Party
+    import torch
+
+    profiles = render_profiles()
+    gpus = 0
+    total_vram = 0
+    train = _train()
+    chosen_vram = None
+    chosen_profile_name = None
+    previous_vram = None
+    # try nvidia
+    if torch.cuda.is_available():
+        click.echo("Detecting Hardware...")
+        gpus = torch.cuda.device_count()
+        for i in range(gpus):
+            properties = torch.cuda.get_device_properties(i)
+            total_vram += properties.total_memory  # memory in B
+    vram = convert_bytes_to_proper_mag(total_vram)[0]
+    for profile in profiles:
+        # for each profile mapping (vram -> list of profiles)
+        # if the vram we have is < the vram for this group of profiles
+        if vram < profile[0]:
+            # for each profile in this vRAM grouping of profiles
+            for profile_names_and_values in profile[1]:
+                if chosen_profile_name is not None:
+                    break
+                # enumerate over the profiles in the list, use .items() so we can access profile names
+                # the structure of this is due to the fact that _train instances cannot be members of a map. they must be in a tuple
+                for profile_name, profile_config in profile_names_and_values.items():
+                    # if our vram is < the vram required for this profile, choose it.
+                    assert isinstance(profile_config["vram"], int)
```

Comment:
Suggestion: Rather than use asserts, if this is a condition we want to guard against, might want to check with an if and print a message and exit if not satisfied